pytest/lib/mocknet.py

```diff
@@ -167,43 +212,59 @@ def chain_measure_bps_and_tps(archival_node,
         end_time = curr_time
     if start_time is None:
         start_time = end_time - duration
+    logger.info(
+        f'Measuring BPS and TPS in the time range {start_time} to {end_time}')
 
-    # one entry per block, equal to the timestamp of that block
+    # One entry per block, equal to the timestamp of that block.
     block_times = []
-    # one entry per block (because there is only one shard), equal to the number of transactions
+    # One entry per block, containing the count of transactions in all chunks of the block.
     tx_count = []
+    block_counter = 0
     while curr_time > start_time:
         if curr_time < end_time:
             block_times.append(curr_time)
-            chunk_hash = curr_block['chunks'][0]['chunk_hash']
-            chunk = archival_node.get_chunk(chunk_hash)['result']
-            tx_count.append(len(chunk['transactions']))
+            txs = 0
+            tx_per_chunk = [None] * len(curr_block['chunks'])
+            pmap(
+                lambda i: get_chunk_txn(i, curr_block['chunks'], archival_node,
+                                        tx_per_chunk),
+                range(len(curr_block['chunks'])))
```

Comment:
Isn’t this just:

    def get_chunk_txn(chunk, archival_node):
        chunk_hash = chunk['chunk_hash']
        chunk = archival_node.get_chunk(chunk_hash)['result']
        return len(chunk['transactions'])

    tx_per_chunk = pmap(
        lambda chunk: get_chunk_txn(chunk, archival_node),
        curr_block['chunks'])

Reply:
Yes and no. You are right that I forgot that `pmap` returns a list. But I also want the results to be in the order of the chunks. To inspect if a single shard gets over- or under-loaded.

Reply:
`pmap` does return values in order.